js/style/style_batch.js

```diff
@@ -46,11 +48,22 @@ styleBatch.prototype = {
         if (this._style._layers[layer.id] !== undefined) {
             throw new Error('There is already a layer with this ID');
         }
+
+
         if (!(layer instanceof StyleLayer)) {
+            if (validateStyle.emitErrors(this._style, validateStyle.layer({
+                value: layer,
+                style: this._style.serialize(),
+                styleSpec: styleSpec
+            }))) return this;
```

Comment:
It seems likes batches should implement consistency, ensuring that either all operations can complete without errors or the style is not modified.